lib/index.js

```diff
@@ -37,7 +45,55 @@ class GoodLoggly extends Stream.Writable {
 
     data.timestamp = GoodLoggly.timeString(data.timestamp);
     data.msg = GoodLoggly.getMessage(data);
-    this._client.log(data, data.tags, callback);
+
+    this._buffer.push(data);
+
+    if (this._bufferReady) {
+      this._sendMessages(callback);
+    } else {
+      if (!this._bufferStart) {
+        this._bufferStart = Date.now();
+      }
+
+      setImmediate(callback);
+    }
+  }
+  get _bufferReady () {
+    if (this._buffer.length >= this._config.threshold) {
+      // Buffer is full
+      return true;
+    }
+
+    if (this._config.maxDelay > 0 && this._bufferStart &&
+      Date.now() - this._bufferStart > this._config.maxDelay) {
+      // Max wait time exceeded
+      return true;
+    }
+
+    return false;
+  }
+  _sendMessages (callback) {
+    const sendComplete = (err) => {
+      this._bufferStart = null;
+      this._buffer = [];
+
+      if (typeof callback === 'function') {
+        callback(err);
+      }
+    };
+
+    if (this._buffer.length) {
+      if (this._buffer.length === 1) {
+        // Backward-compatible behavior to do single log send
+        return this._client.log(this._buffer[0], this._buffer[0].tags, sendComplete);
+      }
+
+      // If using bulk, individual message tags can't be sent
+      return this._client.log(this._buffer, sendComplete);
+    }
+
+    // Fall-through behavior for empty buffer, ensuring async response
+    process.nextTick(() => { sendComplete(null); });
```

Comment:
Unless you have a good reason, `setImmediate` is preferred to `process.nextTick()`